package.json

```diff
@@ -42,10 +42,12 @@
     "defu": "^6.1.4",
     "mime": "^3.0.0",
     "node-fetch-native": "^1.6.4",
-    "pathe": "^1.1.2"
+    "pathe": "^1.1.2",
+    "punycode": "^2.3.1"
```

Comment:
Here is a well-tested and optimized version of punycode that we can inline in ununv: 

https://github.com/unjs/ufo/blob/main/src/punycode.ts

since ufo is part of unjs ecosystem, we could also export utils from ufo directly because this way most of unjs powered builds (including nitro and nuxt) would reuse same bits and save bundle. PR welcome for this but in the meantime, inlining would be better than introducing this dep.

Reply:
Sounds good. Just want to clarify a few things:

- When you say inline -- do you mean actually inline by copy pasting the source of that file into unenv, or just subbing `ufo` for the `punycode.js` dependency here and re-exporting the relevant functions
- Does `ufo` provide sufficient coverage? I see `encodeHost` uses the `toASCII` function from the punycode file, but I don't see anything for decoding, and none of the `ufo` decode functions seem to be able to decode punycode urls (just testing with the ones from the node docs, ex `punycode.decode('--dqo34k'); // '☃-⌘' `)

Reply:
@pi0 lmk what you'd like to see done here. I'm still not sure what to do about `decode` and `ucs2` if we switch to `ufo` since it looks like there isn't sufficient coverage.

Reply:
Added inlined version of punycode with minimal changes and esm compact fixes let's see later how we can minimize it (with proper testing)